Configuration.cpp

```diff
@@ -123,6 +122,31 @@ bool Configuration::readConfigFile(char *path) {
     }
   }
 
+  /*Parsing Continents*/
+   if(!root["continents"].empty()) {
+      if(root["continents"]["whitelist"].empty()) {
+        trace->traceEvent(TRACE_INFO, "Missing %s from %s", "whitelist", path);
+      } else {
+        for(Json::Value::ArrayIndex i = 0; i != root["continents"]["whitelist"].size(); i++) {
+  	std::string continent = root["continents"]["whitelist"][i].asString();
+    continents[country2u16((char*)continent.c_str())] = MARKER_PASS;
+  	trace->traceEvent(TRACE_INFO, "Adding %s to continents whitelist", continent.c_str());
+
+        }
+      }
+
+      if(root["continents"]["blacklist"].empty()) {
+        trace->traceEvent(TRACE_INFO, "Missing %s from %s", "blacklist", path);
+      } else {
+        for(Json::Value::ArrayIndex i = 0; i != root["continents"]["blacklist"].size(); i++) {
+  	std::string continent = root["continents"]["blacklist"][i].asString();
+    continents[country2u16((char*)continent.c_str())] = MARKER_DROP;
+  	trace->traceEvent(TRACE_INFO, "Adding %s to continents blacklist", continent.c_str());
+
+        }
+      }
+    }
+
```

Comment:
Refactor code in order to merge country and continent logic and remove code duplication